src/blob/index/bptree/core.rs

```diff
@@ -376,46 +383,51 @@ where
         Ok(())
     }
 
-    async fn validate_header(&self, buf: &mut Vec<u8>, blob_size: u64) -> Result<()> {
+    async fn validate_header(&self, buf: BytesMut, blob_size: u64) -> Result<BytesMut> {
```

Comment:
This function calls `hash_valid` so here is also better to take `buf` as mutable reference (`buf: &mut BytesMut`)

Reply:
Or even better: to prevent resize we can take it as a slice: `buf: &mut [u8]`